File: solutions/python/difference-of-squares/1/difference_of_squares.py
"""
Finding the difference between the square of the sum 
and the sum of the squares of the first N numbers.
"""

def square_of_sum(number):
    return sum([*range(1, number + 1)]) ** 2

def sum_of_squares(number):
    return sum(map(lambda x: x * x, [*range(1, number + 1)]))

def difference_of_squares(number):
    return square_of_sum(number) - sum_of_squares(number)

File: solutions/fsharp/difference-of-squares/1/DifferenceOfSquares.fs
module DifferenceOfSquares

let squareOfSum (number: int): int = (List.sum [1..number]) * (List.sum [1..number])

let sumOfSquares (number: int): int = List.sumBy (fun x -> x * x) [1..number]

let differenceOfSquares (number: int): int = squareOfSum number - sumOfSquares number